web/src/components/Terminal.tsx

```diff
@@ -0,0 +1,480 @@
+                setTimeout(() => {
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput ])
+                }, 200)
+                setTimeout(() => {
+                    setCurrentCommand('vim code.tsx')
+                    setCurrentCommandIndex(0)
+                }, 1200)
+            } else if (currentCommand === 'vim code.tsx') {
+                setTimeout(() => {
+                    setMode('vim')
+                    setCurrentCommand('')
+                    setLinesChanged(1)
+                }, 200)
+            } else if (currentCommand === 'git commit -m "Minor change to language."') {
+                setTimeout(() => {
+                    setCommandSpeed('low')
+                    setLinesChanged(0)
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput, gitCommitOutput ])
+                }, 200)
+                setTimeout(() => {
+                    setCurrentCommand('git push')
+                    setCurrentCommandIndex(0)
+                }, 1000)
+            } else if (currentCommand === 'git push') {
+                setTimeout(() => {
+                    setCurrentCommand('')
+                    setCommandOutputs([ lsOutput, gitCommitOutput, gitPushOutput ])
+                }, 200)
+            }
+        }
+    }, [ currentCommand, currentCommandCursorIndex ])
+
+    return (
+        mode === 'vim' ? (
+            <VimCodeView animationFinished={vimModeFinished} />
+        ) : (
+            <div className='flex flex-col w-full'>
+                <div className='p-2 flex flex-col justify-end h-72 overflow-hidden'>
+                    {commandOutputs.map(output => (
+                        <pre key={output}>
+                            <span className='font-thin text-slate-200'>
+                                {output}
+                            </span>
+                        </pre>
+                    ))}
+                </div>
+                <Prompt command={currentCommand} commandCursorIndex={currentCommandCursorIndex} linesChanged={linesChanged} />
+            </div>
+        )
+    )
+}
+
+const Prompt = ({ command, commandCursorIndex, linesChanged }: { command: string, commandCursorIndex: number, linesChanged: number }) => {
+    const [ showCursor, setShowCursor ] = useState(true)
+    const cursorInterval = useRef(null)
+
+    const clearCursorInterval = () => {
+        if (cursorInterval.current) {
+            clearInterval(cursorInterval.current)
+        }
+    }
+
+    const createCursorInterval = () => {
+        cursorInterval.current = setInterval(() => {
+            setShowCursor(showCursor => !showCursor)
+        }, 400)
+    }
+
+    useEffect(() => {
+        createCursorInterval()
+        return clearCursorInterval
+    }, [])
+
+    useEffect(() => {
+        setShowCursor(true)
+        clearCursorInterval()
+        createCursorInterval()
+        return clearCursorInterval
```

Comment:
Hell ya to cleaning up that interval like this

Reply:
> A pause button would be dope for the animation I think. It's just fun to pause and be like "oh what was that change he just made in that file?" and kind of analyze the code

With the way that it's implemented, this would definitely be a bitch to implement. lol Maybe it can better be done with some kind of animation library though 🤔.